error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/StringRules.java

```diff
@@ -65,19 +67,20 @@ Optional<String> before(String str) {
 
     @AfterTemplate
     Optional<String> after(String str) {
-      return Optional.ofNullable(str).filter(s -> !s.isEmpty());
+      return Optional.ofNullable(str).filter(not(String::isEmpty));
     }
   }
 
   static final class FilterEmptyString {
     @BeforeTemplate
     Optional<String> before(Optional<String> optional) {
-      return optional.map(Strings::emptyToNull);
+      return Refaster.anyOf(optional.map(Strings::emptyToNull), optional.filter(s -> !s.isEmpty()));
```

Comment:
Now the rule is very specific to this case. Instead we can introduce a separate rule like we have for `EqualsPredicate`. (In general we always try to go the smallest/most generic rule.)